### ISSUE TITLE
Project 3 Feedback

### ISSUE THREAD

**Issue by cameronjacoby** [TheEddyChang/alldaynba#1]
## Project Strengths
- Creative problem solving by accessing NBA data via their in-house API. And nice job on not getting shut down in the process. :basketball: 
- Great use of a rake task to run dynamic API queries based on the day and store that data into your database.
- Excellent work on the `calculate_probability!` method. It's fairly impressive to dive into your own data analysis of any kind.
- Nice use of `scope` to pull `todays_games` from the database!
## Growth Opportunities
- **Project Planning:** Next time you're working on a project, make sure to plan out details beforehand, like which API data you need and how you'll define your database relationships. It helps to start with clear, precise user stories that define the value proposition, and wireframes that depict how users will interact with your site.
- **Naming Conventions:** Your naming is all over the place. You can't control the format of API params or data, but ALWAYS use `snake_case` in Rails for any variable names, method definitions, etc. that you have control over.
- **Primary/Foreign Keys:** You're using the "NBA ids" as primary and foreign keys, and that can be really confusing to other developers. Stick to the Active Record ids unless you have a compelling reason to do otherwise.
- **Routes & Controller Methods:**
  - [These do the same thing](https://github.com/TheEddyChang/alldaynba/blob/master/config/routes.rb#L3-L5) - pick one.
  - [This controller method doesn't exist.](https://github.com/TheEddyChang/alldaynba/blob/master/config/routes.rb#L7)
  - [games#show](https://github.com/TheEddyChang/alldaynba/blob/master/app/controllers/games_controller.rb#L9-L13) isn't very RESTful, since it appears to be an alternate view for `@allGames`. The `show` action should always relate to a single instance of any given resource.
  - [Remove any controller methods you're not using.](https://github.com/TheEddyChang/alldaynba/blob/master/app/controllers/games_controller.rb#L15-L19)
- **Views:** The [random spacing in the views](https://github.com/TheEddyChang/alldaynba/blob/master/app/views/games/index.html.erb#L9-L45) is very confusing, and it makes the code hard to follow.
- **Testing:** You have an excellent method on the `Game` model for calculating probability, and it deserves a test. :smile: 
- **To Get Ready for Deploy:**
  - **README:** Add a `README` file explaining a little about your project, the technologies you used, and how another developer could run the project locally. Follow the <a href="https://github.com/sf-wdi-24/project-03#completed-project-deliverables" target="_blank">Project 3 Guidelines</a>.
  - **Gem Groups:** The `bootstrap-sass` gem is in the `development` group. Make sure to move it out, so the gem can work in production.
  - **Postgres Database:** Change your database from SQLite to Postgres. You can follow <a href="https://devcenter.heroku.com/articles/getting-started-with-rails4" target="_blank">this Heroku guide</a>.
  - **Schedule Rake Task:** You'll need to schedule your `todays_games_info` rake task to run every day on Heroku. Read about the <a href="https://devcenter.heroku.com/articles/scheduler" target="_blank">Heroku Scheduler</a>.
- **User Interactivity:** You did an awesome job sorting through API data and displaying it in a meaningful way. The main thing I see that's missing from this app is an element of user interactivity. Push yourself to dive deeper into Rails by implementing users. Sign up / log in could be optional, but maybe it unlocks additional features, like "watching" specific games or the ability for users to make their own win predictions.

**Comment by cameronjacoby** [comment]
Hi @TheEddyChang - Here is your Project 3 feedback. Let me know if you have any questions about it! I encourage you to keep working on the project, as it's a great start, and I can tell it's something you're passionate about and excited to work on.

@zoejf - Feel free to add any other comments you have!

**Comment by zoejf** [comment]
@TheEddyChang I second everything that Cameron said. You worked with a really tough API and I was really impressed by the perseverance you showed when you hit the ups and downs that came with it. 

I especially agree with Cameron's comments about code cleanup and consistency - this should be your priority. Now that you have a little distance from the intensity of project week, I'd recommend that you try redoing some of your planning documents. This will help you figure out how each component relates to others and what you can get rid of. Lastly, I think the other thing that you should be doing is practice talking about the project and specifically how the code works. Especially being able to talk about what specific lines of the rake task do, as well as the probability calculations. Show your code to a classmate, or even someone who has no coding background and practice explaining things at a basic level.
